docs/go-configuration.md

```diff
@@ -4,27 +4,366 @@ title: Scope Go Agent configuration
 sidebar_label: Configuration
 ---
 
-The behaviour of Scope Agent can be modified using  environment variables.
+The behaviour of Scope Agent can be modified either by code or using environment variables.
+
+Configuration by code is supported for both `scopeagent.Run(m)` and `agent.NewAgent()` as agent options parameters.
 
 > If these properties are manually configured, use `true` or `false` only for the boolean value.
 
 
+## Changing service name
+
+You can specify the name of the service when sending data to Scope. Scope uses `default` as fallback.
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Environment Variable-->
+
+```sh
+SCOPE_SERVICE="service-name"
+```
+
+<!--By Code-->
```

Comment:
```suggestion
<!--Runtime instrumentation-->
```

and I'd also add the example in `Testing instrumentation` using `scopeagent.Run(m)`

Reply:
Ok